io/net/net_tools.py

```diff
@@ -0,0 +1,359 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+#
+# Based on code by Hong Bo Peng <penghb@cn.ibm.com>
+# copyright: 2003, 2015 IBM Corp
+
+import os
+import re
+import avocado
+from avocado import Test
+from avocado import main
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+def install_dependencies():
+    """
+    To check and install dependencies for the test
+    """
+    detected_distro = distro.detect()
+    sm = SoftwareManager()
+    if detected_distro.name == "SuSE":
+        net_tools = ("net-tools", "traceroute")
+    else:
+        net_tools = ("net-tools", "hostname", "traceroute")
+    for pkg in net_tools:
+        if not sm.check_installed(pkg) and not sm.install(pkg):
+            raise AssertionError("%s package is need to test" % pkg)
+
+
+class Hostname(Test):
+    """
+    hostname: set/get hostname by cmd. Check it with the expected result.
+    Test to verify the functionality of the hostname command
+    """
+
+    def setUp(self):
+        install_dependencies()
+        self.restore_hostname = False
+        # Get Hostname
+        hostname = process.system_output("hostname").strip('\n')
+        if not hostname:
+            # set hostname if not set
+            process.system("hostname localhost.localdomain", sudo=True)
+            hostname = "localhost.localdomain"
+        self.hostname = hostname
+
+    @avocado.fail_on(process.CmdError)
+    def test_hostname(self):
+        """
+        i.Verifies hostname command options and
+        ii.Test to change the hostname
+        """
+        output = process.system_output("hostname")
+        if not output:
+            self.fail("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            ret = process.run("hostname -%s" % option, ignore_status=True)
+            if ret.exit_status:
+                self.fail("Hostname reported non-zero status %s for option %s"
+                          % (ret.exit_status, option))
+            if not ret.stdout:
+                self.fail("No output for %s option" % (option))
+
+        # Test to change hostname
+        myhostname_file = os.path.join(self.workdir, "MYHOSTNAME")
+        myhostname = "myhost.my-domain"
+        if myhostname == self.hostname:
+            myhostname += '1'
+        with open(myhostname_file, 'w') as fobj:
+            fobj.write(myhostname)
+
+        process.system("hostname -F %s" % myhostname_file,
+                       sudo=True)
+        self.restore_hostname = True
```

Comment:
You should set the `restore_hostname` before executing the `hosname -F`, because if the command changes the name and crashes, you don't get to restore it... (basically please swap the lines 89 and 91)